### PR TITLE
Patching windows.groovy to execute on a helix queue with .net 472

### PR DESCRIPTION
**Pull request by JohnTortugo**
Resolves dotnet/core-eng#3607

### CODE REVIEW

**Comment by MattGal** [comment]
This definitely  used the right Helix queue but from this log, it would seem it's looking for an unset environment variable:

```
14:53:50 D:\j\workspace\windows-TGrou---74aa877a>"\VsDevCmd.bat"   && msbuild src\upload-tests.proj /p:TargetGroup=netcoreapp /p:ArchGroup=x64 /p:ConfigurationGroup=Debug /p:TestProduct=corefx /p:TimeoutInSeconds=1200 /p:TargetOS=Windows_NT ...
```

I recommend tracking down what it's constructing the path with VsDevCmd with .

**Comment by JohnTortugo** [comment]
windows.groovy assumes %VS140COMMONTOOLS% exist and VSDevCmd.bat is there:

https://github.com/dotnet/corefx/blob/0c6cb66794c04a29030b2512673e240c2da98a87/buildpipeline/windows.groovy#L85

which as far as I understood by reading some comments means that it implies that VS 14 is installed:

https://github.com/dotnet/corefx/blob/28a593f4257f74791539553012b62c70adff018a/run.cmd#L7

Which version of VS is installed on this queue?

**Comment by MattGal** [comment]
Dev15.  (VS 15). It's the flavor all the official builds are using now anyways.

**Comment by MattGal** [comment]
@JohnTortugo : Work with the product team.  I don't think they specifically need VS 2014.  I favor option 4 on your list "just fix the environment variable" with using VSWhere as a close second.

**Comment by JohnTortugo** [comment]
@safern do you have any strict requirement on VS14? 

It seems that calling VsDevCmd like this `%_VSCOMNTOOLS%\\VsDevCmd.bat` is the more "portable" way if you don't have a strict requirement.

**Comment by JohnTortugo** [comment]
@dotnet-bot test this please

File: buildpipeline/windows.groovy [hunk 1]
```diff
@@ -82,7 +82,9 @@ simpleNode('Windows_NT','latest') {
                     targetHelixQueues = ['Windows.10.Amd64.ClientRS4.Open']
                 }
 
-                bat "\"%VS140COMNTOOLS%\\VsDevCmd.bat\" && msbuild src\\upload-tests.proj /p:TargetGroup=${params.TGroup} /p:ArchGroup=${params.AGroup} /p:ConfigurationGroup=${params.CGroup} /p:TestProduct=corefx /p:TimeoutInSeconds=1200 /p:TargetOS=Windows_NT /p:HelixJobType=test/functional/cli/ /p:HelixSource=${helixSource} /p:BuildMoniker=${helixBuild} /p:HelixCreator=${helixCreator} /p:CloudDropAccountName=dotnetbuilddrops /p:CloudResultsAccountName=dotnetjobresults /p:CloudDropAccessToken=%CloudDropAccessToken% /p:CloudResultsAccessToken=%OutputCloudResultsAccessToken% /p:HelixApiEndpoint=https://helix.dot.net/api/2017-04-14/jobs /p:TargetQueues=\"${targetHelixQueues.join(',')}\" /p:HelixLogFolder= /p:HelixLogFolder=${WORKSPACE}\\${logFolder}\\ /p:HelixCorrelationInfoFileName=SubmittedHelixRuns.txt"
+                bat ".\\setup-vs-tools.cmd";
```
Comment (weshaggard) [inline buildpipeline/windows.groovy]: nit: you can do this in one line like the older call to VSDevCmd

**Comment by JohnTortugo** [comment]
@MattGal @weshaggard sounds good to merge?

**Comment by MattGal** [comment]
Looks good to me, and you actually got a green run.

**Comment by JohnTortugo** [comment]
Sure. Created this other PR: https://github.com/dotnet/corefx/pull/31286

**Pull request merged**